### PR TITLE
OK, this should be a bit cleaner

### PR DESCRIPTION
**Pull request by Satsuoni**
So, I tried to clean up the code a bit. I won't have time to fill out all of the functions, and so I'll stop here, but for the moment:
* FPGA code cleaned up and folded, so fpga_nfc bit becomes unnecessary
* felica.c has functions for Tag simulation and reader simulation, and modulating/demodulating packets
* client has sample functions for felica snooping (with some annotations) and FelicaLiteS dump
Cons:
* I have been experimenting with various additional settings to existing modes. Some are still there. Existing settings should all be untouched.
* While your fork compiles, my proxmark refuses to be recognized afterwards, so I am not sure I integrated all the functionality into your branch correctly. (This is also true for your master branch to begin with) 
* Please don't merge your client Makefile <_<
Anyway, hopefully this will be a bit more clear.  As before, please feel free to just pick whatever you need.

### CODE REVIEW

**Comment by iceman1001** [comment]
Nice work!
Its a big PR, will still take time to go through. 
Now what is up with the client/makefile ?!?

**Comment by Satsuoni** [comment]
I removed the QT detection, since it didn't work on my config (I seem to have several Qt versions installed on my Windows, and the make gets confused) And then forgot to put it back before committing.

File: armsrc/fpgaloader.h [hunk 1]
```diff
@@ -85,4 +85,14 @@ extern void switch_off();
 #define FPGA_HF_ISO14443A_READER_LISTEN				(3<<0)
 #define FPGA_HF_ISO14443A_READER_MOD				(4<<0)
 
+
+//options for Felica. 
+#define FPGA_MAJOR_MODE_ISO18092          			(5<<5)
+#define FPGA_HF_ISO18092_FLAG_NOMOD                 (1<<0) //disable modulation module
+#define FPGA_HF_ISO18092_FLAG_414K                  (2<<0) // should enable 414k mode (untested). No autodetect
```
Comment (Satsuoni) [inline armsrc/fpgaloader.h:92]: This should be 424k

File: client/Makefile [hunk 2]
```diff
@@ -42,19 +42,7 @@ endif
...
-ifeq ($(QTINCLUDES), )
-# if Qt5 not found check for correctly configured Qt4	
-	QTINCLUDES = $(shell pkg-config --cflags QtCore QtGui 2>/dev/null)
-	QTLDLIBS = $(shell pkg-config --libs QtCore QtGui 2>/dev/null)
-	MOC = $(shell pkg-config --variable=moc_location QtCore)
-	UIC = $(shell pkg-config --variable=uic_location QtCore)
-else
-	CXXFLAGS += -std=c++11 -fPIC
-endif
+
```
Comment (Satsuoni) [inline client/Makefile:45]: This part should not be removed on properly configured system, I think.

**Comment by iceman1001** [comment]
OK,  I can't merge this one either.  
I have however added all of it already.
You would need to fix up the code a bit. its quite messy.

**Pull request closed without merging**